### PR TITLE
Fixes #1290 When loading an organ, individual parameters are not created

### LINKED ISSUE #1290: When loading an organ, individual parameters are not created.
1. Load a PK-Sim model in MoBi
2. Export an organ as PKML and select an individual to create the parameters from
3. Create a new module and load the organ in this module
4. None of the individual-specific parameters are created:

image

### PR DESCRIPTION
**Pull request by rwmcintosh**
Fixes #1290

# Description
This brings back code that exports ```IndividualParameters``` in the container itself when exporting a container with individual and expression. ```ExpressionParameters``` are still exported as ```ParameterValues```

## Type of change

Please mark relevant options with an `x` in the brackets.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires documentation changes (link at least one [user](https://github.com/Open-Systems-Pharmacology/docs) or [developer](https://github.com/Open-Systems-Pharmacology/developer-docs) documentation issue):
- [ ] Algorithm update - updates algorithm documentation/questions/answers etc.
- [ ] Other (please describe):

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Integration tests
- [x] Unit tests
- [x] Manual tests 
- [ ] No tests required

# Reviewer checklist

Mark everything that needs to be checked before merging the PR.

- [ ] Check if the code is well documented
- [ ] Check if the behavior is what is expected
- [ ] Check if the code is well tested
- [ ] Check if the code is readable and well formatted
- [ ] Additional checks (document below if any)
- [ ] Check if documentation update issue(s) are created if the option `This change requires a documentation update` above is selected

# Screenshots (if appropriate):

# Questions (if appropriate):

### CODE REVIEW

File: src/MoBi.Presentation/Tasks/Edit/EditTaskForContainer.cs [hunk 1]
```diff
@@ -86,6 +94,66 @@ public override void Save(IContainer container)
          exportContainer(container, fileName);
       }
 
+      private void addIndividualParametersToContainers(IndividualBuildingBlock individual, IContainer container)
```
Comment (Yuri05) [inline src/MoBi.Presentation/Tasks/Edit/EditTaskForContainer.cs:97]: would rename to `addIndividualParametersToContainer` (singular)
(and before rename the current `addIndividualParametersToContainer` to `addIndividualParametersToContainerByPath` or something like that)

File: src/MoBi.Presentation/Tasks/Edit/EditTaskForContainer.cs [hunk 2]
```diff
@@ -86,6 +94,66 @@ public override void Save(IContainer container)
...
+            addIndividualParametersToContainer(individual, subContainer, new ObjectPath(subContainerPath));
+         });
+      }
+
+      private void addIndividualParametersToContainer(IndividualBuildingBlock individual, IContainer container, ObjectPath containerPath)
+      {
+         var individualParametersToExport = individual.Where(individualParameter => individualParameter.ContainerPath.StartsWith(containerPath)).ToList();
+
+         // Ensure that the distributed parameters are added to the container first. After all distributed parameters are added, then the other parameters can be added.
+         individualParametersToExport.Where(x => x.DistributionType != null).Each(x => addIndividualParameterToContainer(x, container, containerPath));
```
Comment (Yuri05) [inline src/MoBi.Presentation/Tasks/Edit/EditTaskForContainer.cs:115]: shouldn't we better use `.Where(x => !x.IsDistributed())` instead?

File: src/MoBi.Presentation/Tasks/Edit/EditTaskForContainer.cs [hunk 3]
```diff
@@ -86,6 +94,66 @@ public override void Save(IContainer container)
...
+      }
+
+      private IParameter createParameter(IndividualParameter individualParameter)
+      {
+         var parameterToAdd = _individualParameterToParameterMapper.MapFrom(individualParameter);
+
+         if (individualParameter.Formula != null)
+            parameterToAdd.Formula = individualParameter.Formula;
+         if (individualParameter.Value.HasValue)
+            parameterToAdd.Formula = _formulaFactory.ConstantFormula(individualParameter.Value.Value, individualParameter.Dimension);
```
Comment (Yuri05) [inline src/MoBi.Presentation/Tasks/Edit/EditTaskForContainer.cs:142]: Questions on this:

a) shouldn't it be encapsulated in `IndividualParameterToParameterMapper.MapFrom()` ?
b) don't we need to **clone** the formula in `parameterToAdd.Formula = individualParameter.Formula;` ?
c) the second assignment overrides the first one (if the condition is true), so I would rather do it as
```
if (individualParameter.Value.HasValue)
...
else if (individualParameter.Formula != null)
...
```

File: src/MoBi.Presentation/Tasks/Edit/EditTaskForContainer.cs (hunk 1 above)
Comment (rwmcintosh) [reply src/MoBi.Presentation/Tasks/Edit/EditTaskForContainer.cs:97]: I'm going with ```addIndividualParametersToContainerAndSubContainers``` for this one and ```addIndividualParametersToContainerByPath ``` for the other

File: src/MoBi.Presentation/Tasks/Edit/EditTaskForContainer.cs (hunk 2 above)
Comment (rwmcintosh) [reply src/MoBi.Presentation/Tasks/Edit/EditTaskForContainer.cs:115]: That does not yet exist for ```PathAndValueEntity``` based entities. I'll add the extension in Core, then add a core update here to use it after you approve the core PR

File: src/MoBi.Presentation/Tasks/Edit/EditTaskForContainer.cs (hunk 3 above)
Comment (rwmcintosh) [reply src/MoBi.Presentation/Tasks/Edit/EditTaskForContainer.cs:142]: Well, now I'm not so sure this should be encapsulated. We are using the same mapper for model creation, so the cloning works differently. I'll put the else-if here and add a comment why this is not encapsulated.

Comment (Yuri05) [reply src/MoBi.Presentation/Tasks/Edit/EditTaskForContainer.cs:142]: ok

**Review by Yuri05 — APPROVED** [review]

**Pull request merged**